### PR TITLE
Add standalone installation

### LINKED ISSUE #8: [Request] Add instructions to run standalone
In some configurations, servers will not be running docker (due to e.g. resource constraints) and won't have systemd (e.g. {Free,Dragonfly,Open}BSD hosts). A good example of this are free services like serv00 that offer a small FreeBSD VPS as well as keeping your site up-and-running. But if you've got a domain elsewhere, or want your site to be accessible as a subdomain of something else, you're gonna need to put in the effort manually. It'd be very nice to have this running as a cronjob in these situations.

In my case I have a VPS elsewhere, but it still runs FreeBSD. And running Docker/Podman just for SSL certs seems a bit overkill - especially since I'm mostly serving some static files from it on the website end of things.

I think a less experimentative user could benefit from instructions on setting up "standalone" running of the Python scripts through cron rather than as a container

### PR DESCRIPTION
**Pull request by RivenSkaye**
Fixes #8 
Shortly after opening it I realized I might as well write it myself. All it really needs is a crontab entry and a wrapper that can turn the .env file into a bunch of environment variables. This is almost that, just needs a small tweak to ignore blank lines and lines starting with a # as these currently make `printf` error

### CODE REVIEW

**Comment by RivenSkaye** [comment]
It now ignores empty lines and comments, as well as any lines where the value itself starts with (a space and) a comment.

I've tried to keep it as OS and cron-agnostic as possible, which means running on Mondays. Matthew Dillon's dcron (now maintained by Jim Pryor) _might_ disagree with day selection 0 in the weekly script on account of different crons using 0, 7 or both for Sundays. I have no idea which it does or doesn't accept these days, so defaulting to day 1 is the most portable solution here.

Jim Pryor did add the `@daily` extension (as well as `@reboot`), but I'm sure someone out there is using an esoteric cron that doesn't like that. This is gonna be as good as it gets.

Please do double check that the sh scripts are fully portable. They should be, but bashisms creep up on everyone

File: README.md [hunk 1]
```diff
@@ -72,6 +72,23 @@ docker compose up
 
 This will create the `porkcron` container and download the certificate bundle into the `ssl` volume.
 
+### Standalone for use with cron
```
Comment (tmzane) [inline README.md]:
```suggestion
### Using cron
```

Wdyt about naming this method just "cron"? Seeing "standalone", I'm thinking that there would be no dependencies, but in the end we still need cron installed. "Using systemd/docker/cron" sounds better to me.

File: standalone/install.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+cp ../porkcron.sh /usr/local/bin/porkcron
+chmod +x /usr/local/bin/porkcron
+
+mkdir -p /etc/porkcron
+cp ../porkcron.py /etc/porkcron/porkcron.py
+chmod +x /etc/porkcron/porkcron.py
+
+# Add a cron entry for it to run on reboot and weekly
+echo '0 0 * * 1 root /usr/local/bin/porkcron' > /etc/cron.d/porkcron.weekly
+echo '@reboot root /usr/local/bin/porkcron' >> /etc/cron.d/porkcron.reboot
```
Comment (tmzane) [inline standalone/install.sh]: Can we move the crontab file from /docker to the root and reuse for both methods instead? 🤔

File: standalone/install.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env sh
```
Comment (tmzane) [inline standalone/install.sh]:
```suggestion
#!/bin/sh
```

IIRC, existing scripts in this repo use /bin/sh, let's make it consistent. Same below

File: standalone/porkcron.sh [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    case $value in
+      \ \#*|\#*) continue ;;
+    esac
+    printf -v "$key" %s "$value"
+    export "$key"
+  done <.env
+else
+  echo "Could not find $PWD/.env to fetch the latest SSL certs!" 1>&2
+  echo "Trying anyway, it should work if the correct environment variables are set elsewhere." 1>&2
+fi
```
Comment (tmzane) [inline standalone/porkcron.sh]: Would a simple
```
export $(grep -v '^#' .env | xargs)
```
work instead? I wouldn't worry about corner cases, our .env file is super simple

**Review by tmzane — COMMENTED** [review]
This looks great, thanks a lot for the contribution! Did you test it on your server? If so, how did it work for you?

A few thoughts from me:

File: README.md (hunk 1 above)
Comment (RivenSkaye) [reply README.md]: Sounds fine to me. I figured cron would be a negligible dependency because of the project name, but I'll push the change

File: standalone/install.sh (hunk 2 above)
Comment (RivenSkaye) [reply standalone/install.sh]: I specifically made it separate output files here due to not all flavors of cron supporting the `@periodic` directives. Old dcron/crond installs wouldn't be able to parse it, cronie [needs to be built to do so](https://bugs.gentoo.org/551352), and I'm not sure what e.g. systemd-cron and similar wrappers/converters do to them (the current AUR version is at least a few versions behind).

That's also why I didn't put them in `/etc/cron.{weekly,reboot}/`. Some cron implementations don't read those by default, `/etc/cron.d/` is the only place they all agree on

File: standalone/porkcron.sh (hunk 4 above)
Comment (RivenSkaye) [reply standalone/porkcron.sh]: I could certainly test it with a clean .env file. Went with this because I needed something to counteract my own stupidity when testing and setting up

**Comment by RivenSkaye** [comment]
I'm using almost this. The only difference is that I manually put the files in the right places and then echo'd lines from history into a script, and then made it more robust and portable with stuff like splitting crontabs and changing the running day to 1 (Monday).

Edit:
Some additional info, I did first run this as root in a WSL env just to check it all works, but I obviously don't have reliable cron in there and did not want to hammer Porkbun's API with anacron features by accident

File: standalone/porkcron.sh (hunk 4 above)
Comment (RivenSkaye) [reply standalone/porkcron.sh]: Update: no it does not. `export` is pretty finicky here and as a shell builtin it also can't be called through `grep ... | xargs export`
Running it over the unmodified example, as well as having comments behind properly filled out lines, causes these kind of errors:
```
gitserver@GiteaPi:~/porkcron$ export $(grep -v '^#' .env.example | xargs)
-bash: export: `#': not a valid identifier
-bash: export: `comma-separated': not a valid identifier
gitserver@GiteaPi:~/porkcron$ export $(grep -v '^\#' .env.example | xargs)
-bash: export: `#': not a valid identifier
-bash: export: `comma-separated': not a valid identifier
```

edit: when making sure all keys are set AND there are no spaces in any keys (e.g. multiple domains cannot be split by `, `) it does seem to work. Might need to document that, or check for spaces in lines/keys. The middle ground would then be blindly exporting line-by-line doing `export "$key='$value'"`

Comment (RivenSkaye) [reply standalone/porkcron.sh]: Ended with almost the suggested solution. Allowing for whitespace before the comment marker, and using a simple sed call to fix unexpected results if any value contains `, ` by shaving off the space.
Purely as an accident, this does allow using spaces for e.g. friendlier softwrapping in editors

File: standalone/install.sh (hunk 2 above)
Comment (RivenSkaye) [reply standalone/install.sh]: Taking a quick peek at the used docker container and I was apparently wrong. Alpine defaults to Busybox crond and [they have yet another handling of cron paths](https://github.com/mirror/busybox/blob/371fe9f71d445d18be28c82a2a6d82115c8af19d/miscutils/crond.c#L54). And the currently specified directory for the crontab doesn't match anything in those sources.  
I could `cat` them together in a single file during the run step, but I'm not sure if that works and I don't have a machine to test that on
